test/failure_detector_tests.ml

```diff
@@ -0,0 +1,49 @@
+open Commons
+open Pollinate.Node
+open Lwt.Infix
+
+module SUT = Pollinate.Node.Failure_detector
+
+let node_a =
+  Lwt_main.run
+    (Node.init ~router:Commons.router ~state:["test1"]
+       ~msg_handler:Commons.msg_handler ("127.0.0.1", 3003))
+let node_b =
+  Lwt_main.run
+    (Node.init ~router:Commons.router ~state:["test1"]
+       ~msg_handler:Commons.msg_handler ("127.0.0.1", 3004))
+
+let peer_b = Client.peer_from !node_b
+
+let suspicion_detection () =
+  let open Common.Peer in
+  let _ = add_neighbor (Pollinate.Node.Client.peer_from !node_a) peer_b in
+  let _ = SUT.update_peer_status node_a peer_b Suspicious in
+  let%lwt _ = SUT.suspicious_detection node_a in
+  Lwt.return (Base.Hashtbl.length !node_a.peers = 0)
```

Comment:
It's nice that we can break away a small bit of functionality and test it this way, but `suspicious_detection` probably will not be called this way in the library, so we should try and later add some integration tests that ensure the suspicion detector is performing as expected when combined with the rest of the library.

Reply:
Totally agreed. This is clearly irrelevant of the targeted way to use this function. But I wanted to make a quick test about its perimeter, in order to be sure the status are well calculated and the Hashtbl is well updated.
I guess it will be reworked in some future integration tests, but it is still enough right now to ensure this function has a correct behavior. Don't you think?